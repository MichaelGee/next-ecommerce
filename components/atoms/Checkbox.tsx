import {useState} from 'react'
import { Checkbox, Label } from 'atomize';


interface ICheckbox {
  text: string;
  checked: boolean;
  onChange: Function;
}

export const CheckboxComponent = (props: ICheckbox) => {
    const { text, checked, onChange } = props;
   
   
   
    return (
        <>
            <Label align="center" textSize='caption' textWeight="500" m={{ b: "0.5rem" }}>
                <Checkbox
                    onChange={onChange}
                    checked={checked}
                    inactiveColor="contrast01"
                    activeColor="brand01"
                    size="23px"
                />
                {text}
            </Label>
        </>
    )
}
