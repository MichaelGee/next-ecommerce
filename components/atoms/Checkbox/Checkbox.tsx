import {useState} from 'react'
import { Checkbox, Label } from 'atomize';


interface ICheckbox {
  text?: string;
  size?: string;
  textSize?: string;
  checked?: boolean;
  onChange?: Function;
}

export const CheckboxComponent = (props: ICheckbox) => {
    const { text, size, textSize, checked, onChange } = props;
   
   
   
    return (
        <>
            <Label align="center" textSize={textSize} textWeight="500" m={{ b: "0.5rem" }}>
                <Checkbox
                    onChange={onChange}
                    checked={checked}
                    inactiveColor="contrast01"
                    activeColor="brand01"
                    size={size}
                />
                {text}
            </Label>
        </>
    )
}
