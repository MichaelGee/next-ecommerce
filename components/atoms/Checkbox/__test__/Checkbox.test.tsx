import '@testing-library/jest-dom'
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import { CheckboxComponent } from '../Checkbox';


configure({ adapter: new Adapter() });


describe('render input', () => {
    it('matches the snapshot', () => {
        const tree = renderer.create(<CheckboxComponent />).toJSON();
        expect(tree).toMatchSnapshot();
    });
    test('check if input component exist', () => {
        const wrapper = shallow(<CheckboxComponent />);
        expect(wrapper.find(<label />));
         expect(wrapper.find(<input />));
    })
})