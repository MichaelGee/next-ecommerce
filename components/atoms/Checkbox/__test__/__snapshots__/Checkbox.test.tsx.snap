// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render input matches the snapshot 1`] = `
<label
  className=""
  tabIndex="0"
>
  <input
    className=""
    color="black"
    cursor="pointer"
    type="checkbox"
  />
  <svg
    className=""
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    viewBox="0 0 24 24"
  >
    <path
      d="M5.8458278,4 C4.92853228,4 4.70418221,4.043324 4.47792916,4.16432545 C4.33757722,4.23938649 4.23938649,4.33757722 4.16432545,4.47792916 C4.043324,4.70418221 4,4.92853228 4,5.8458278 L4,18.1541722 C4,19.0714677 4.043324,19.2958178 4.16432545,19.5220708 C4.23938649,19.6624228 4.33757722,19.7606135 4.47792916,19.8356745 C4.70418221,19.956676 4.92853228,20 5.8458278,20 L18.1541722,20 C19.0714677,20 19.2958178,19.956676 19.5220708,19.8356745 C19.6624228,19.7606135 19.7606135,19.6624228 19.8356745,19.5220708 C19.956676,19.2958178 20,19.0714677 20,18.1541722 L20,5.8458278 C20,4.92853228 19.956676,4.70418221 19.8356745,4.47792916 C19.7606135,4.33757722 19.6624228,4.23938649 19.5220708,4.16432545 C19.2958178,4.043324 19.0714677,4 18.1541722,4 L5.8458278,4 Z M5.8458278,2 L18.1541722,2 C19.4914503,2 19.9763797,2.13923842 20.4652686,2.40069906 C20.9541574,2.66215969 21.3378403,3.04584256 21.5993009,3.53473144 C21.8607616,4.02362033 22,4.50854969 22,5.8458278 L22,18.1541722 C22,19.4914503 21.8607616,19.9763797 21.5993009,20.4652686 C21.3378403,20.9541574 20.9541574,21.3378403 20.4652686,21.5993009 C19.9763797,21.8607616 19.4914503,22 18.1541722,22 L5.8458278,22 C4.50854969,22 4.02362033,21.8607616 3.53473144,21.5993009 C3.04584256,21.3378403 2.66215969,20.9541574 2.40069906,20.4652686 C2.13923842,19.9763797 2,19.4914503 2,18.1541722 L2,5.8458278 C2,4.50854969 2.13923842,4.02362033 2.40069906,3.53473144 C2.66215969,3.04584256 3.04584256,2.66215969 3.53473144,2.40069906 C4.02362033,2.13923842 4.50854969,2 5.8458278,2 Z"
      fill="contrast01"
      fillRule="evenodd"
    />
  </svg>
</label>
`;
